chain/events/filter/index.go

```diff
@@ -520,18 +655,26 @@ func (ei *EventIndex) prefillFilter(ctx context.Context, f *eventFilter, exclude
 		}
 	}
 
-	if excludeReverted {
-		clauses = append(clauses, "event.reverted=?")
-		values = append(values, false)
+	// Resolve emitters from filter addresses at the latest TipSet.
+	// TODO investigate if it is safe to use the current tipset; what about finality?
+	emitters, err := f.emitters(ctx, nil)
+	if err != nil {
+		return xerrors.Errorf("emitters clauses: %w", err)
 	}
-
-	if len(f.addresses) > 0 {
-		subclauses := []string{}
-		for _, addr := range f.addresses {
-			subclauses = append(subclauses, "emitter_addr=?")
-			values = append(values, addr.Bytes())
+	if len(emitters) > 0 {
+		subclauses := make([]string, 0, len(emitters))
+		for _, emitter := range emitters {
+			subclauses = append(subclauses, "emitter=?")
+			values = append(values, emitter)
 		}
 		clauses = append(clauses, "("+strings.Join(subclauses, " OR ")+")")
+		// Explicitly exclude reverted events, since at least one emitter is present and reverts cannot be considered.
+		excludeReverted = true
```

Comment:
What does this mean ? Why does having atleast one matching emitter with the filter lead us to exclude reverted events ?

Reply:
Because we cannot confidently resolve the address for reverted events.

Reply:
We should just always exclude reverted events when prefilling (for now). The option to not exclude them only exists for testing.

I've filed an issue. https://github.com/filecoin-project/lotus/issues/11770

Reply:
I think until we've done proper work to resolve #11770 (see thread there) you'd have to _include_ reverted even in the case of having emitters to consider. I think you need to give maximal information to someone seeking to do a look-back because they're likely trying to catch up _unless_ they exclude reverted themselves in which case they are not doing a "tell me what happened since I last looked" catch-up but a "tell me how it is now".

Reply:
The issue with that is address resolution; events database may contain reverted events without emitter actor ID. Therefore, we cannot confidently infer whether those events should be included in pre-filling or not. Does that make sense?

Reply:
Well, after the migration, they'll have IDs, they just won't have addresses. But you're right that we won't be able to confidently pre-fill (quickly, at least).

I'm going to have to disagree with @rvagg here. Our handling of reverted events in the current APIs is not very useful, as discussed in the linked issue.

Reply:
Progressed with always excluding reverted events during backfill. Please mark as unresolved if there is anything that I have missed here. thanks